src/google/google-calendars/googlecalendarsyncadaptor.cpp

```diff
@@ -623,8 +624,14 @@ void GoogleCalendarSyncAdaptor::purgeDataForOldAccount(int oldId, SocialNetworkS
 void GoogleCalendarSyncAdaptor::beginSync(int accountId, const QString &accessToken)
 {
     SOCIALD_LOG_DEBUG("beginning Calendar sync for Google, account" << accountId);
-
     bool needCleanSync = !wasLastSyncSuccessful(accountId);
+    if (needCleanSync) {
```

Comment:
So, I do this here (and removed the similar code from the updateLocalCalendarNotebookEvents() function, which used to have some cleanup-for-cleanSync handling) because the m_storage has no changes at this point in time; so we can do the deletions and then call m_storage->save() (inside finaliseCleanup() due to CleanUpPurge flag).

However, for some reason, it doesn't appear to work - and I cannot figure out why!